gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -58,7 +60,9 @@ async fn main() -> Result<(), anyhow::Error> {
 
     Server::builder()
         .add_service(GatewayLightningServer::new(service))
-        .serve(listen)
+        .serve_with_shutdown(listen, async {
+            let _ = tg.make_handle().make_shutdown_rx().await.await;
```

Comment:
Could you add a comment about what is happening here?

It seems like we're waiting for a shutdown signal to fire. And that returns the function? Does that stop the server? Unclear without looking up documentation ...